src/core/dbus-unit.c

```diff
@@ -782,14 +782,15 @@ static int bus_unit_method_freezer_generic(sd_bus_message *message, void *userda
         if (r == 0)
                 reply_no_delay = true;
 
-        assert(!u->pending_freezer_message);
+        if (u->pending_freezer_message) {
+                bus_unit_send_pending_freezer_message(u, true);
+                assert(!u->pending_freezer_message);
+        }
 
-        r = sd_bus_message_new_method_return(message, &u->pending_freezer_message);
-        if (r < 0)
-                return r;
+        u->pending_freezer_message = sd_bus_message_ref(message);
```

Comment:
Hmm, this is changing semantics of what this variable holds, maybe it should be renamed accordingly so it is a bit less confusing, e.g. `freezer_request` or so.

Reply:
Done, renamed to `pending_freezer_invocation`.